model/src/ww3_ounf.F90

```diff
@@ -0,0 +1,4112 @@
+        WRITE (IDDDAY,'(I10,1X)') INT(DTREQ/86400.)
+      ELSE
+        IDDDAY = '           '
+      END IF
+      IDTIME(1:11) = IDDDAY
+      IDTIME(21:23) = '   '
+      WRITE (NDSO,941) IDTIME, NOUT
+
+      IF(FLGFC) THEN
+        CALL STME21 ( TREF , IDTIME )
+        WRITE(NDSO,942) IDTIME
+      ENDIF
+
+! 4.2 Output fields
+      DO IFI=1, NOGRP
+        DO IFJ=1, NGRPP
+          IF ( FLG2D(IFI,IFJ) ) THEN
+            IF ( FLOGRD(IFI,IFJ) ) THEN
+              WRITE (NDSO,946) IDOUT(IFI,IFJ), ' '
+            ELSE
+              WRITE (NDSO,946) IDOUT(IFI,IFJ), '*** NOT AVAILABLE ***'
+              FLG2D(IFI,IFJ) = .FALSE.
+            END IF
+          END IF
+        END DO
+      END DO
+
+
+! 4.3 Output type
+      ALLOCATE(TABIPART(NOSWLL + 1))
+      ALLOCATE(NCIDS(NOGRP,NGRPP,NOSWLL + 1))
+      NBIPART=0
+      DO I=1,30
+        IF(STRINGIPART(I:I) .EQ. ' ') CYCLE
+        READ(STRINGIPART(I:I),'(I1)') IPART
+        IF(IPART .GT. NOSWLL) THEN
+           WRITE(NDSO, 1500) IPART, NOSWLL
+           CYCLE
+        ENDIF
+        NBIPART = NBIPART + 1
+        IF(NBIPART .GT. NOSWLL + 1) THEN
+           GOTO 803
+        ENDIF
+        TABIPART(NBIPART) = IPART
+      ENDDO
+!
+      IF ( NCTYPE.LT.3 .OR. NCTYPE.GT.4 ) THEN
+        WRITE (NDSE,1010) NCTYPE
+        CALL EXTCDE ( 1 )
+      END IF
+      ! if NCTYPE = 4 checking that it is compiled with NC4
+      HASNC4=0
+#ifdef W3_NC4
+      HASNC4=1
+#endif // W3_NC4
+      IF ((HASNC4 .EQ. 0).AND.(NCTYPE.EQ.4)) THEN
+        WRITE (NDSE,1012)
+        CALL EXTCDE ( 1 )
+      END IF
+
+#ifdef W3_SMC
+      WRITE(NDSO, 4100)
+      IF(SMCTYPE .EQ. 1) THEN  ! Flat sea point output
+        ALLOCATE(SMCMASK(NSEA))
+        ALLOCATE(SMCIDX(NSEA))
+        SMCMASK(:) = .FALSE.
+        CALL SMC_INTERP()
+        SMCNOUT = COUNT(SMCMASK)
+        NXO = SMCNOUT
+        NYO = 1
+        WRITE(NDSO, 4120) SMCNOUT
+      ELSE IF(SMCTYPE .EQ. 2) THEN  ! Regular grid output
+        ! Calculate regridding weights:
+        ALLOCATE(XIDX(NSEA), YIDX(NSEA), XSPAN(NSEA),                   &
+                 YSPAN(NSEA), WTS(NSEA), SMCIDX(NSEA))
+        CALL SMC_INTERP()
+        WRITE(NDSO, 4110) NXO, NYO, SXO, SYO, DXO, DYO
+
+        ! Allocate space for coverage array and new MAPSTA array
+        ALLOCATE(COV(NXO,NYO), MAPSMC(NXO,NYO))
+      ELSE IF(SMCTYPE .EQ. 3 .OR. SMCTYPE .EQ. 4) THEN  ! Nearest neighbour interpolation
+        CALL READ_SMCINT()
+      ENDIF
+
+      ! CB: IXN and IXY are calculated by SMC_INTERP for SMC GRID
+      IX1 = 1
+      IXN = NXO
+      IY1 = 1
+      IYN = NYO
+     
+      ! Also store NXO and NYO in __local__ RTDNX and RTDNY variables.
+      ! This avoids compilation errors when the RTD switch is enabled
+      ! but the SMC switch is not. TODO: Remove this when C-preprocessor
+      ! is used in preference to switches.
+      RTDNX = NXO
+      RTDNY = NYO
+
+!/RTD      ! SMC type 3/4 outputs are currently on standard pole grid only
```

Comment:
@JessicaMeixner-NOAA One thing I did not factor into my awk conversion script is the conversion of lines that contain double switches, like this one (the second `!/RTD` switch is still present).

E.g. lines that look something lie this:
`!/SMC!/RTD   ! This is a double switch line`

In these cases, to avoid making the awk script overly complicated, I found it easier to simply rerun the conversion script until it reports no more converted switches.

Reply:
You could add an extra loop in your control script to test for the existence of any switches in the input file, something like this (unteted!) around line 18 of `ftn2src.sh`:

```
.
.
       echo "convert $file.ftn to a .F90"   
       cp -f $ftndir/${file}.ftn $ftndir/${file}.tmp
       while $(egrep "^\!/[0-9A-Za-z]+" ${file}.tmp); do
          gawk -f $thisdir/switch2cpp.awk < $ftndir/${file}.ftn > $srcdir/${file}.F90
          cp -f $ftndir/${file}.F90 $ftndir/${file}.tmp
       done
       rm -f  $ftndir/${file}.tmp
.
.
```

Reply:
@ukmo-ccbunney  thanks for catching the double switch issue! I'll add a second loop to take care of this.  Is there any issue in just running the switch2cpp.awk on every file twice?

Reply:
>  Is there any issue in just running the switch2cpp.awk on every file twice?

I don't think so - it should simply report "**Converted 0 switches...**" for most files that don't have double switches.
Now - I don't think there any triple switches...but maybe we should check :)

Reply:
Okay, I've updated for processing 3 times to be extra cautious.  I also updated this PR to show the changes in the ftn directory because I think that's easier to see changes then too.